port/unix/omrsock.c

```diff
@@ -134,14 +163,14 @@ omrsock_getaddrinfo_socktype(struct OMRPortLibrary *portLibrary, omrsock_addrinf
 }
 
 int32_t
-omrsock_getaddrinfo_protocol(struct OMRPortLibrary *portLibrary, omrsock_addrinfo_t handle, int32_t *protocol, int32_t index)
+omrsock_getaddrinfo_protocol(struct OMRPortLibrary *portLibrary, omrsock_addrinfo_t handle, int32_t *protocol, uint32_t index)
 {
 	if ((NULL == handle) || (NULL == handle->addrInfo) || (index < 0) || (index >= handle->length)) {
```

Comment:
From the compiler warning, index is never less than 0.
```suggestion
	if ((NULL == handle) || (NULL == handle->addrInfo) || (index >= handle->length)) {
```

Reply:
Sounds good, they have been fixed.